xcq-extension/procedural/src/lib.rs

```diff
@@ -165,11 +165,14 @@ fn replace_self_to_impl(ty: &syn::Type) -> syn::Result<Box<syn::Type>> {
     Ok(Box::new(modified_ty))
 }
 
-// TODO: may rely on whole syn::File where we can replace the type alias to get stable hash results
-// Or we rejects the type alias
-// TODO: check hasher implementation is collision resistant and stable
-fn calculate_hash(trait_items: &[TraitItem]) -> u64 {
-    let mut hasher = std::hash::DefaultHasher::new();
-    std::hash::Hash::hash_slice(trait_items, &mut hasher);
+// TODO: currently we only hash on trait ident and function names,
+fn calculate_hash(trait_ident: &Ident, trait_items: &[TraitItem]) -> u64 {
+    let mut hasher = twox_hash::XxHash64::default();
+    trait_ident.hash(&mut hasher);
+    for trait_item in trait_items {
+        if let TraitItem::Fn(method) = trait_item {
+            method.sig.ident.hash(&mut hasher);
+        }
+    }
```

Comment:
```suggestion
    let mut hasher = twox_hash::XxHash64::default();
    trait_ident.hash("xcq-trait$");
    trait_ident.hash(&mut hasher);
    for trait_item in trait_items {
        if let TraitItem::Fn(method) = trait_item {
            trait_ident.hash("@");
            method.sig.ident.hash(&mut hasher);
        }
    }
```

to reduce the chance of hash collision

Reply:
How adding two static string improve the hash collision?

Reply:
otherwise chose will result the same thing

```rs
trait Test {
  func a()
  func b()
}

trait Test {
  func ab()
}

trait Testa {
  func b()
}
```

Reply:
it is also generally good idea to salt the hash payload https://en.wikipedia.org/wiki/Salt_(cryptography)